packages/@aws-cdk/aws-appsync/lib/schema-intermediate.ts

```diff
@@ -296,3 +296,102 @@ export class InputType implements IIntermediateType {
     this.definition[options.fieldName] = options.field;
   }
 }
+
+/**
+ * Properties for configuring an Union Type
+ *
+ * @props definition - the object types for this union type
+ *
+ * @experimental
+ */
+export interface UnionTypeProps {
+  /**
+   * the object types for this union type
+   */
+  readonly definition: IIntermediateType[];
+}
+
+/**
+ * Union Types are abstract types that are similar to Interface Types,
+ * but they don't get to specify any common fields between types.
+ *
+ * Note that fields of a union type need to be object types. In other words,
+ * you can't create a union type out of interfaces, other unions, or inputs.
+ *
+ * @experimental
+ */
+export class UnionType implements IIntermediateType {
+  /**
+   * the name of this type
+   */
+  public readonly name: string;
+  /**
+   * the attributes of this type
+   */
+  public readonly definition: { [key: string]: IField };
+  /**
+   * the authorization modes for this intermediate type
+   */
+  protected modes?: AuthorizationType[];
+
+  public constructor(name: string, props: UnionTypeProps) {
+    this.name = name;
+    this.definition = {};
+    props.definition.map((def) => this.addField({ field: def.attribute() }));
+  }
+
+  /**
+   * Create an GraphQL Type representing this Union Type
+   *
+   * @param options the options to configure this attribute
+   * - isList
+   * - isRequired
+   * - isRequiredList
+   */
```

Comment:
nit: I'm not sure this annotation adds much that intellisense/auto complete would not already offer. it feels redundant. wdyt?

Reply:
yeah i agree we can just remove